sycl/source/backend.cpp

```diff
@@ -202,14 +202,21 @@ make_kernel_bundle(pi_native_handle NativeHandle, const context &TargetContext,
   return make_kernel_bundle(NativeHandle, TargetContext, false, State, Backend);
 }
 
-kernel make_kernel(pi_native_handle NativeHandle, const context &TargetContext,
+kernel make_kernel(const context &TargetContext,
+                   const kernel_bundle<bundle_state::executable> &KernelBundle,
+                   pi_native_handle NativeHandle, bool KeepOwnership,
                    backend Backend) {
   const auto &Plugin = getPlugin(Backend);
   const auto &ContextImpl = getSyclObjImpl(TargetContext);
+  const auto &DeviceImage = *KernelBundle.begin();
+  const auto &DeviceImageImpl = getSyclObjImpl(DeviceImage);
```

Comment:
1. Please, replace `auto` with actual types.
2. Please add a comment clarifying why taking the first `device_image` is OK. Maybe it makes sense to have an assert checking that the number of `device_image's is 1.